block/meta-iosched.c

```diff
@@ -288,6 +320,10 @@ void update_stats(struct request_queue *q) {
 	whole_stat++;
 	spin_unlock(&stat_lock);
 	q->stats[group]++;
+
+	//spin_lock(q->queue_lock); //spin_lock_irq(q->queue_lock);
```

Comment:
Also doing q->stats[2]++ under spinlock looks wasteful, you could drop all locking and use atomics here.